src/Http/Http.Results/src/ObjectResult.cs

```diff
@@ -61,7 +66,7 @@ public Task ExecuteAsync(HttpContext httpContext)
             }
 
             OnFormatting(httpContext);
-            return httpContext.Response.WriteAsJsonAsync(Value);
+            return httpContext.Response.WriteAsJsonAsync(Value, Value.GetType(), options: null, contentType: ContentType);
```

Comment:
Ugh, callsites like this are what make trimming hard. I guess we'll need new result types and overloads to fix the problem.

Reply:
> I guess we'll need new result types and overloads

Sorry, I don't super follow.

Reply:
To allow for annotating things using the DAM attributes we either need a Type argument to flow or a generic argument to flow. That means ObjectResult needs a Type argument or it needs to be generic (ObjectResult\<T\>). Then you would mark those with the attributes to make sure stuff doesn't get trimmed but maybe this is premature.